R/mbbs.R

```diff
@@ -130,6 +130,37 @@ create_route_level_counts_0 <- function(ebird, stop_level_data, taxonomy, config
         })()
```

Comment:
> Can't comment on line 118 directly, but just to double check, what's the goal with filtering to only routes without stop_num?

This function is creating the route level data, and those ebird checklists with stop information will come from the `stop_level` dataset.

Reply:
Ok, so that filter makes then. So... my proposed solutions are to run the left_join check for differences before line 122-130 where all the ebird observations that are already included in any part of the stop_level dataset are removed. When I do that there's 691 differences out of 4,850 records with overlap.

When I add 'source' as part of the grouping in 110/the select in 112, I start getting many-to-many join errors in the left_join. Is there already a check for overlaps from different sources with creating the stop_level data? I see there's a logger in create_stop_level_counts_0, but not they're not removed. I think that's likely where the fix lies eg: fix stop_level overlaps -> add 'source' to the route-level creation grouping so any remaining errors can be tracked down more easily -> the logger that starts at 133 will be able to catch genuine differences between stop level and route level.
BC there's not really 691 differences or I would have caught them (I hope _OTL)  

For this pull request, do you want to end it at making sure the logic works (by moving up where the function that checks for differences is), or I can start kicking around the overlap entries (eg. check if they're the same, if the same, remove one, otherwise if differences THEN add note to the logger)

Reply:
Answered my own question! I'll push the changes to fix the logic for the pr and then open another one for the overlap entries :)